zrml/court/src/lib.rs

```diff
@@ -708,16 +715,18 @@ mod pallet {
 
             match draws.binary_search_by_key(&who, |draw| draw.court_participant.clone()) {
                 Ok(index) => {
-                    let draw = draws[index].clone();
-
-                    // allow to override last vote
-                    ensure!(
-                        matches!(draws[index].vote, Vote::Drawn | Vote::Secret { commitment: _ }),
-                        Error::<T>::InvalidVoteState
-                    );
+                    if let Some(draw) = draws.get_mut(index) {
```

Comment:
I know that `debug_assert!` statements are an integral part of how you write code. I'd just like to point out a key difference to what I would recommend, which is:

```
let draw = draws.get_mut(index).unwrap_or_else(|| Error::<T>::Unexpected);
```

The way this code is now written, you are absolutely relying on `draws.get_mut` returning `Some`. If it fails to do so in production, this function will produce unexpected results. That's definitely better than panicking, but be aware that you're actually taking a risk here.